lib/repositories/switch-repository.ts

```diff
@@ -0,0 +1,78 @@
+import { DynamoDBDocumentClient } from '@aws-sdk/lib-dynamodb';
+import Logger from 'bunyan';
+import { Entity, Table } from 'dynamodb-toolbox';
+
+import { DYNAMO_TABLE_KEY, DYNAMO_TABLE_NAME } from '../constants';
+import { SynthSwitchRequestBody } from '../handlers/synth-switch';
+import { BaseSwitchRepository } from './base';
+
+export const PARTITION_KEY = `${DYNAMO_TABLE_KEY.INPUT_TOKEN}#${DYNAMO_TABLE_KEY.INPUT_TOKEN_CHAIN_ID}#${DYNAMO_TABLE_KEY.OUTPUT_TOKEN}#${DYNAMO_TABLE_KEY.OUTPUT_TOKEN_CHAIN_ID}#${DYNAMO_TABLE_KEY.TRADE_TYPE}`;
+
+export class SwitchRepository implements BaseSwitchRepository {
+  static log: Logger;
+
+  static create(documentClient: DynamoDBDocumentClient): BaseSwitchRepository {
+    this.log = Logger.createLogger({
+      name: 'DynamoSwitchRepository',
+      serializers: Logger.stdSerializers,
+    });
+
+    const switchTable = new Table({
+      name: DYNAMO_TABLE_NAME.SYNTH_SWITCH,
+      partitionKey: PARTITION_KEY,
+      sortKey: `${DYNAMO_TABLE_KEY.LOWER}`,
+      DocumentClient: documentClient,
+    });
+
+    const switchEntity = new Entity({
+      name: 'SynthSwitchEntity',
+      attributes: {
+        [PARTITION_KEY]: { partitionKey: true },
+        lower: { sortKey: true },
+        enabled: { type: 'boolean' },
+      },
+      table: switchTable,
+      autoExecute: true,
+    } as const);
+
+    return new SwitchRepository(switchTable, switchEntity);
+  }
+
+  private constructor(
+    // eslint-disable-next-line
+    private readonly _switchTable: Table<
+      'SynthSwitch',
+      'inputToken#inputTokenChainId#outputToken#outputTokenChainId#type',
```

Comment:
```suggestion
      PARTITION_KEY,
```
maybe?

Reply:
yeah unfortunately PARTITION_KEY is a value so cannot be used as a type parameter here (at least that's what the linter says)